pkg/patch/patch.go

```diff
@@ -337,6 +326,34 @@ func patchWithContext(ctx context.Context, ch chan error, image, reportFile, pat
 	return err
 }
 
+// resolvePatchedTag merges explicit tag & suffix rules, returning the final patched tag.
+func resolvePatchedTag(imageRef reference.Named, explicitTag, suffix string) (string, error) {
+	// if user explicitly sets a final tag, that wins outright
+	if explicitTag != "" {
+		return explicitTag, nil
+	}
+
+	// parse out any existing tag from the image ref
+	var baseTag string
+	if tagged, ok := imageRef.(reference.Tagged); ok {
+		baseTag = tagged.Tag()
+	}
+
+	// if suffix is empty, default to "patched"
+	if suffix == "" {
+		suffix = "patched"
+	}
+
+	// if we have no original baseTag (the user’s image had no tag),
+	// then final name is just the suffix
+	if baseTag == "" {
+		return suffix, nil
+	}
```

Comment:
could this return in an error if the original image is just `test` and the patched image becomes `test:-patched`?